pkg/controller/llmisvc/controller.go

```diff
@@ -96,36 +96,40 @@ func (r *LLMInferenceServiceReconciler) Reconcile(ctx context.Context, req ctrl.
 	resource := original.DeepCopy()
 
 	reconciler.PreProcessReconcile(ctx, resource)
-	reconcileErr := r.ReconcileKind(ctx, resource)
+	reconcileErr := r.reconcile(ctx, resource)
 	reconciler.PostProcessReconcile(ctx, resource, original)
 
-	if err := r.updateStatus(ctx, original, resource); err != nil {
-		return ctrl.Result{}, err
-	}
-
 	if reconcileErr != nil {
 		logger.Error(reconcileErr, "Failed to reconcile LLMInferenceService")
-		r.Recorder.Eventf(original, corev1.EventTypeWarning, "Error", reconcileErr.Error())
-		return ctrl.Result{}, reconcileErr
+		r.Recorder.Eventf(original, corev1.EventTypeWarning, "Error", "Reconciliation failed: %v", reconcileErr.Error())
+	} else {
+		logger.Info("Reconciliation completed successfully")
+		r.Recorder.Eventf(original, corev1.EventTypeNormal, "Reconciled", "LLMInferenceService reconciled successfully.")
```

Comment:
This is something we got bit by in the past, producing an event on success (but also on no-op) can cause a high load on the API server.

In addition, a non-error doesn't mean the object is ready / configured correctly, so the message might even be missleading.

Reply:
Reference https://github.com/knative/pkg/issues/1520